File: backend/src/main/java/az/project/VocabularyBuilder/vocabulary/dto/VocabularyEntryView.java
package az.project.VocabularyBuilder.vocabulary.dto;

import az.project.VocabularyBuilder.vocabulary.domain.ContextSource;
import az.project.VocabularyBuilder.vocabulary.domain.MasteryStatus;
import az.project.VocabularyBuilder.vocabulary.domain.PartOfSpeech;

import java.time.LocalDateTime;

public record VocabularyEntryView(
        Long id,
        String wordPhraseEn,
        String translationPl,
        PartOfSpeech partOfSpeech,
        MasteryStatus masteryStatus,
        Integer correctAnswerStreak,
        Integer totalCorrectAnswers,
        LocalDateTime createdAt,
        ContextSource contextSource,
        String sourceTitle,
        Integer episodeNumber,
        Integer timeOffsetSeconds
) {
}

File: frontend/src/app/features/vocabulary-quiz/models/vocabularyEntryView.ts
import { ContextSource } from "../../../shared/enums/contextSource";
import { MasteryStatus } from "../../../shared/enums/masterStatus";
import { PartOfSpeech } from "../../../shared/enums/partOfSpeech";

export interface VocabularyEntryView {
  id: number; // Zmienione z Long na number dla JavaScript
  wordPhraseEn: string;
  translationPl: string;
  partOfSpeech: PartOfSpeech;
  masteryStatus: MasteryStatus;
  contextSource: ContextSource;
  correctAnswerStreak: number; // Zmienione z Integer na number
  totalCorrectAnswers: number; // Zmienione z Integer na number
  createdAt: Date; // Zmienione z LocalDateTime na Date
  sourceTitle: string;
  episodeNumber: number | null;
  timeOffsetSeconds: number | null;
  imageUrl: string | null;
}